allennlp/common/file_utils.py

```diff
@@ -1085,20 +1085,27 @@ def get_file_extension(path: str, dot=True, lower: bool = True):
 
 
 def open_compressed(
-    filename: Union[str, PathLike], mode: str = "rt", encoding: Optional[str] = "UTF-8", **kwargs
+    filename: Union[str, PathLike],
+    compression_type: str = None,
```

Comment:
This should be the last parameter, so we don't break existing usage of positional arguments.

Reply:
Also, the type of this should be `Optional[None]`.

Reply:
the `make typecheck` commands gave me incompatibility error on changing it to `Optional[None]` so I kept it as `Optional[str]`, will that be fine?